liferay-content-management/liferay-portal
# LPD-25954 | LPD-26178 Size filter in Content Dashboard

LPD-25954 and LPD-26178

# What is this trying to solve?

[Link to ticket](https://issues.liferay.com/browse/LPD-25954)
[Link to ticket](https://issues.liferay.com/browse/LPD-26178)

The PR adds 2 changes:

1. Adds the option of having a submenu in the dropdown items for the toolbar
2. Adds a new filter to the content dashboard for the size

# How am I fixing it?

Clay Taglib was refactored so that it now includes submenus. Content dashboard has been refactored to that now the filters are contributed dinamically. And a new filter for size was added

# How can you verify that it works?

The PR includes 3 types of tests

1. A playwright test to check that the a submenu is opened for size
2. An unit test to check that the filters of a filter provider are taken into account when building the search context
3. An integration test to check that the filters for size are working correctly

The PR can also be tested through the following steps

1. Add a new file with size less than or equals to 150KB
2. Add a new file with size between 150KB and 1MB
3. Add a new file with size larger than 1MB
4. Go to Content Dashboard
5. In the toolbar, select size > Small
6. The file smaller than 150KB will be shown
7. In the toolbar, select size > Medium
8. The file between 150KB and 1MB will be shown
9. In the toolbar, select size > Large
10. The file larger than 1MB will be shown
